sycl/test/esimd/simd_view.cpp

```diff
@@ -117,6 +117,29 @@ SYCL_ESIMD_FUNCTION void bar() {
   foo(v0.select<8, 1>(0)); // rvalue
 }
 
+// This test checks that simd_view_impl APIs return objects of derived class.
+// I.e. they never return objects of simd_view_impl class.
+void test_simd_view_impl_api_ret_types() SYCL_ESIMD_FUNCTION {
+  simd<float, 4> x = 0;
+  auto v1 =
+      x.select<2, 1>(0); // simd_view<simd<float, 4>, region1d_t<float, 2, 1>>
+  static_assert(detail::is_simd_view_v<decltype(v1)>::value, "");
+  auto v2 = v1.select<1, 1>(
```

Comment:
Could you also add a case to test v2.bit_cast_view?

Reply:
Done